diff --git a/tinygrad/runtime/autogen/hip.py b/tinygrad/runtime/autogen/hip.py
index b2e1a7de..7ba49008 100644
--- a/tinygrad/runtime/autogen/hip.py
+++ b/tinygrad/runtime/autogen/hip.py
@@ -206,6 +206,96 @@ struct_hipUUID_t._fields_ = [
 ]
 
 hipUUID = struct_hipUUID_t
+class struct_c__SA_hipDeviceArch_t(Structure):
+    pass
+
+struct_c__SA_hipDeviceArch_t._pack_ = 1 # source:False
+struct_c__SA_hipDeviceArch_t._fields_ = [
+    ('hasGlobalInt32Atomics', ctypes.c_uint32, 1),
+    ('hasGlobalFloatAtomicExch', ctypes.c_uint32, 1),
+    ('hasSharedInt32Atomics', ctypes.c_uint32, 1),
+    ('hasSharedFloatAtomicExch', ctypes.c_uint32, 1),
+    ('hasFloatAtomicAdd', ctypes.c_uint32, 1),
+    ('hasGlobalInt64Atomics', ctypes.c_uint32, 1),
+    ('hasSharedInt64Atomics', ctypes.c_uint32, 1),
+    ('hasDoubles', ctypes.c_uint32, 1),
+    ('hasWarpVote', ctypes.c_uint32, 1),
+    ('hasWarpBallot', ctypes.c_uint32, 1),
+    ('hasWarpShuffle', ctypes.c_uint32, 1),
+    ('hasFunnelShift', ctypes.c_uint32, 1),
+    ('hasThreadFenceSystem', ctypes.c_uint32, 1),
+    ('hasSyncThreadsExt', ctypes.c_uint32, 1),
+    ('hasSurfaceFuncs', ctypes.c_uint32, 1),
+    ('has3dGrid', ctypes.c_uint32, 1),
+    ('hasDynamicParallelism', ctypes.c_uint32, 1),
+    ('PADDING_0', ctypes.c_uint16, 15),
+]
+
+hipDeviceArch_t = struct_c__SA_hipDeviceArch_t
+class struct_hipDeviceProp_t(Structure):
+    pass
+
+struct_hipDeviceProp_t._pack_ = 1 # source:False
+struct_hipDeviceProp_t._fields_ = [
+    ('name', ctypes.c_char * 256),
+    ('totalGlobalMem', ctypes.c_uint64),
+    ('sharedMemPerBlock', ctypes.c_uint64),
+    ('regsPerBlock', ctypes.c_int32),
+    ('warpSize', ctypes.c_int32),
+    ('maxThreadsPerBlock', ctypes.c_int32),
+    ('maxThreadsDim', ctypes.c_int32 * 3),
+    ('maxGridSize', ctypes.c_int32 * 3),
+    ('clockRate', ctypes.c_int32),
+    ('memoryClockRate', ctypes.c_int32),
+    ('memoryBusWidth', ctypes.c_int32),
+    ('totalConstMem', ctypes.c_uint64),
+    ('major', ctypes.c_int32),
+    ('minor', ctypes.c_int32),
+    ('multiProcessorCount', ctypes.c_int32),
+    ('l2CacheSize', ctypes.c_int32),
+    ('maxThreadsPerMultiProcessor', ctypes.c_int32),
+    ('computeMode', ctypes.c_int32),
+    ('clockInstructionRate', ctypes.c_int32),
+    ('arch', hipDeviceArch_t),
+    ('concurrentKernels', ctypes.c_int32),
+    ('pciDomainID', ctypes.c_int32),
+    ('pciBusID', ctypes.c_int32),
+    ('pciDeviceID', ctypes.c_int32),
+    ('maxSharedMemoryPerMultiProcessor', ctypes.c_uint64),
+    ('isMultiGpuBoard', ctypes.c_int32),
+    ('canMapHostMemory', ctypes.c_int32),
+    ('gcnArch', ctypes.c_int32),
+    ('gcnArchName', ctypes.c_char * 256),
+    ('integrated', ctypes.c_int32),
+    ('cooperativeLaunch', ctypes.c_int32),
+    ('cooperativeMultiDeviceLaunch', ctypes.c_int32),
+    ('maxTexture1DLinear', ctypes.c_int32),
+    ('maxTexture1D', ctypes.c_int32),
+    ('maxTexture2D', ctypes.c_int32 * 2),
+    ('maxTexture3D', ctypes.c_int32 * 3),
+    ('PADDING_0', ctypes.c_ubyte * 4),
+    ('hdpMemFlushCntl', ctypes.POINTER(ctypes.c_uint32)),
+    ('hdpRegFlushCntl', ctypes.POINTER(ctypes.c_uint32)),
+    ('memPitch', ctypes.c_uint64),
+    ('textureAlignment', ctypes.c_uint64),
+    ('texturePitchAlignment', ctypes.c_uint64),
+    ('kernelExecTimeoutEnabled', ctypes.c_int32),
+    ('ECCEnabled', ctypes.c_int32),
+    ('tccDriver', ctypes.c_int32),
+    ('cooperativeMultiDeviceUnmatchedFunc', ctypes.c_int32),
+    ('cooperativeMultiDeviceUnmatchedGridDim', ctypes.c_int32),
+    ('cooperativeMultiDeviceUnmatchedBlockDim', ctypes.c_int32),
+    ('cooperativeMultiDeviceUnmatchedSharedMem', ctypes.c_int32),
+    ('isLargeBar', ctypes.c_int32),
+    ('asicRevision', ctypes.c_int32),
+    ('managedMemory', ctypes.c_int32),
+    ('directManagedMemAccessFromHost', ctypes.c_int32),
+    ('concurrentManagedAccess', ctypes.c_int32),
+    ('pageableMemoryAccess', ctypes.c_int32),
+    ('pageableMemoryAccessUsesHostPageTables', ctypes.c_int32),
+]
+
+hipDeviceProp_t = struct_hipDeviceProp_t
 class struct_hipDeviceProp_tR0600(Structure):
     pass
 
@@ -2657,9 +2747,9 @@ try:
 except AttributeError:
     pass
 try:
-    hipGetDevicePropertiesR0600 = _libraries['libamdhip64.so'].hipGetDevicePropertiesR0600
-    hipGetDevicePropertiesR0600.restype = hipError_t
-    hipGetDevicePropertiesR0600.argtypes = [ctypes.POINTER(struct_hipDeviceProp_tR0600), ctypes.c_int32]
+    hipGetDeviceProperties = _libraries['libamdhip64.so'].hipGetDeviceProperties
+    hipGetDeviceProperties.restype = hipError_t
+    hipGetDeviceProperties.argtypes = [ctypes.POINTER(struct_hipDeviceProp_t), ctypes.c_int32]
 except AttributeError:
     pass
 try:
@@ -2713,7 +2803,7 @@ except AttributeError:
 try:
     hipChooseDeviceR0600 = _libraries['libamdhip64.so'].hipChooseDeviceR0600
     hipChooseDeviceR0600.restype = hipError_t
-    hipChooseDeviceR0600.argtypes = [ctypes.POINTER(ctypes.c_int32), ctypes.POINTER(struct_hipDeviceProp_tR0600)]
+    hipChooseDeviceR0600.argtypes = [ctypes.POINTER(ctypes.c_int32), ctypes.POINTER(struct_hipDeviceProp_t)]
 except AttributeError:
     pass
 try:
@@ -5441,7 +5531,7 @@ __all__ = \
     'hipDeviceGetUuid', 'hipDeviceGraphMemTrim', 'hipDeviceP2PAttr',
     'hipDevicePrimaryCtxGetState', 'hipDevicePrimaryCtxRelease',
     'hipDevicePrimaryCtxReset', 'hipDevicePrimaryCtxRetain',
-    'hipDevicePrimaryCtxSetFlags', 'hipDeviceProp_tR0600',
+    'hipDevicePrimaryCtxSetFlags', 'hipDeviceProp_t',
     'hipDeviceReset', 'hipDeviceSetCacheConfig',
     'hipDeviceSetGraphMemAttribute', 'hipDeviceSetLimit',
     'hipDeviceSetMemPool', 'hipDeviceSetSharedMemConfig',
@@ -5857,7 +5947,7 @@ __all__ = \
     'struct_hipAccessPolicyWindow', 'struct_hipArray',
     'struct_hipArrayMapInfo', 'struct_hipArrayMapInfo_1_miptail',
     'struct_hipArrayMapInfo_1_sparseLevel',
-    'struct_hipChannelFormatDesc', 'struct_hipDeviceProp_tR0600',
+    'struct_hipChannelFormatDesc', 'struct_hipDeviceProp_t',
     'struct_hipExtent', 'struct_hipExternalMemoryBufferDesc_st',
     'struct_hipExternalMemoryHandleDesc_st',
     'struct_hipExternalMemoryHandleDesc_st_0_win32',
@@ -5905,5 +5995,5 @@ __all__ = \
     'union_hipExternalSemaphoreSignalParams_st_0_nvSciSync',
     'union_hipExternalSemaphoreWaitParams_st_0_nvSciSync',
     'union_hipKernelNodeAttrValue', 'union_hipResourceDesc_res']
-hipDeviceProp_t = hipDeviceProp_tR0600
-hipGetDeviceProperties = hipGetDevicePropertiesR0600
+hipDeviceProp_t = hipDeviceProp_t
+hipGetDeviceProperties = hipGetDeviceProperties
